ee/server/service/users.go

```diff
@@ -0,0 +1,63 @@
+package service
+
+import (
+	"context"
+	"errors"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+)
+
+// GetSSOUser is the premium implementation of svc.GetSSOUser, it allows to
+// create users during the SSO flow the first time they log in if
+// config.SSOSettings.EnableJITProvisioning is `true`
+func (svc *Service) GetSSOUser(ctx context.Context, auth fleet.Auth) (*fleet.User, error) {
+	config, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting app config")
+	}
+
+	if auth.UserID() == "" {
+		return nil, ctxerr.New(ctx, "missing user identifier in SSO response")
+	}
+
+	user, err := svc.Service.GetSSOUser(ctx, auth)
+	var nfe fleet.NotFoundError
+	switch {
+	case err == nil:
+		return user, nil
+	case errors.As(err, &nfe):
+		if !config.SSOSettings.EnableJITProvisioning {
+			return nil, err
+		}
+	default:
+		return nil, err
+	}
+
+	displayName := auth.UserDisplayName()
+	if displayName == "" {
+		displayName = auth.UserID()
+	}
+
+	user, err = svc.Service.NewUser(ctx, fleet.UserPayload{
+		Name:       &displayName,
+		Email:      ptr.String(auth.UserID()),
+		SSOEnabled: ptr.Bool(true),
+		GlobalRole: ptr.String(fleet.RoleObserver),
+	})
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "creating new SSO user")
+	}
+	err = svc.ds.NewActivity(
+		ctx,
+		user,
+		fleet.ActivityTypeUserAddedBySSO,
```

Comment:
Was the UI updated to handle this new type?

Reply:
Ah yes, just found it.

Reply:
thanks for double checking, it was in the related [PR](https://github.com/fleetdm/fleet/pull/7189/files#diff-c1da986073dbf9d1f81c25ae0009896f573f9a2054c13160519a8e58dcd604f4R54-R56) !